pkg/cloudevents/sink_publisher.go

```diff
@@ -0,0 +1,103 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cloudevents
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+
+	ce "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/kubearchive/kubearchive/pkg/constants"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/rest"
+	kclient "knative.dev/eventing/pkg/client/clientset/versioned"
+)
+
+type SinkCloudEventPublisher struct {
+	client client.Client
+	target string
+	source string
+	etype  string
+}
+
+func NewSinkCloudEventPublisher(source string, etype string) (*SinkCloudEventPublisher, error) {
+	scep := &SinkCloudEventPublisher{source: source, etype: etype}
+
+	var err error
+	if scep.client, err = ce.NewClientHTTP(); err != nil {
+		slog.Error("Failed to create client", "error", err)
+		return nil, err
+	}
```

Comment:
I think you should create an instrumented cloudevent http client here. You'll have to import the instrumentation library and the cloudevent httpProtocol packages

``` 
otelObs "github.com/cloudevents/sdk-go/observability/opentelemetry/v2/client"
cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
```

```suggestion
	if scep.client, err = otelObs.NewClientHTTP([]cehttp.Option{}, []client.Option{}); err != nil {
		slog.Error("Failed to create client", "error", err)
		return nil, err
	}
```

Reply:
Fixed.